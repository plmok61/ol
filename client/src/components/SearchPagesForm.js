import React from 'react'
import { Button } from 'react-bootstrap'

const SearchPagesForm = ({ loadBusinesses }) => {
  const handleSearch = (event) => {
    event.preventDefault()
    const page = parseInt(document.getElementById('pageSearch').value)
    const perPage = parseInt(document.getElementById("perPage").value)
    if (isNaN(page)|| page < 1) {
      alert('Please enter a number greater than 1')
    } else {
      loadBusinesses(`http://ec2-54-84-251-148.compute-1.amazonaws.com/businesses/?page=${page}&per_page=${perPage}`)
    }
  }

  return (
    <form onSubmit={handleSearch} className="page-search-form">
      <label>Page #</label>
      <input type="text" id="pageSearch" />
      <label>Results per page</label>
      <select id="perPage">
        <option value="50">50</option>
        <option value="25">25</option>
        <option value="10">10</option>
      </select>
      <Button type="submit" bsSize="small" bsStyle="primary">Submit</Button>
    </form>
  )
}

export default SearchPagesForm