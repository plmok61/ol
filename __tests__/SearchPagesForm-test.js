import React from 'react'
import SearchPagesForm from '../client/src/components/SearchPagesForm'
import renderer from 'react-test-renderer'

test('SearchPagesForm renders', () => {
  const component = renderer.create(
    <SearchPagesForm />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})