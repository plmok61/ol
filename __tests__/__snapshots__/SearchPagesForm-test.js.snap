exports[`test SearchPagesForm renders 1`] = `
<form
  className="page-search-form"
  onSubmit={[Function]}>
  <input
    id="pageSearch"
    placeholder="Page number"
    type="text" />
  <select
    id="perPage">
    <option
      value="50">
      50
    </option>
    <option
      value="25">
      25
    </option>
    <option
      value="10">
      10
    </option>
  </select>
  <input
    type="submit"
    value="submit" />
</form>
`;
